storage/range_test.go

```diff
@@ -1592,7 +1592,7 @@ func TestRangeStatsComputation(t *testing.T) {
 	if err := tc.rng.AddCmd(proto.Put, pArgs, pReply, true); err != nil {
 		t.Fatal(err)
 	}
-	expMS = engine.MVCCStats{LiveBytes: 116, KeyBytes: 30, ValBytes: 86, IntentBytes: 24, LiveCount: 2, KeyCount: 2, ValCount: 2, IntentCount: 1}
+	expMS = engine.MVCCStats{LiveBytes: 118, KeyBytes: 30, ValBytes: 88, IntentBytes: 24, LiveCount: 2, KeyCount: 2, ValCount: 2, IntentCount: 1}
```

Comment:
The encodings are compatible, the generated marshaling is handling the presence of `nullable = false` fields differently than the reflecting based marshaling for custom types. In particular, the encoding of `proto.Key` in `proto.KeyValue` differs. With the generated marshaling, `proto.KeyValue{}` encodes to `0a 00 12 00`. With the reflecting based marshaling it encodes to `12 00`. The `0a 00` portion corresponds to the encoding of the Key field. 

This is a bug in gogoproto, but it it is an efficiency bug, not a correctness one.